lib/avro/builder/types/record_type.rb

```diff
@@ -5,7 +5,10 @@ module Types
       # at the top-level or as the type for a field in a record.
       class RecordType < Avro::Builder::Types::NamedType
 
-        dsl_attributes :doc
+        DSL_METHODS = Set.new(%i(required optional extends)).freeze
```

Comment:
It's a matter of preference but you can do `%i(required optional extends).to_set.freeze` to avoid a level of nesting in favor of chaining.